app/controllers/posts_controller.rb

```diff
@@ -9,13 +9,19 @@ def index
   end
 
   # GET /posts/1
-  def show; end
+  def show
+    if @post
```

Comment:
What if post not present for delete and destroy?
Why you check it only for show?